### PR TITLE
Reintroducing PV objects on demand instead of on startup...

### PR DESCRIPTION
**Pull request by lisazacarias**
- Introducing support for abort
- introducing worker status class for guis that use threads

### CODE REVIEW

File: lcls_tools/common/pydm_tools/displayUtils.py [hunk 1]
```diff
@@ -1,17 +1,49 @@
...
+        super().__init__()
+        if label:
+            self.status.connect(label.setText)
+            self.status.connect(partial(label.setStyleSheet, STATUS_STYLESHEET))
+            
+            self.finished.connect(label.setText)
+            self.finished.connect(partial(label.setStyleSheet, FINISHED_STYLESHEET))
+            
+            self.error.connect(label.setText)
+            self.error.connect(partial(label.setStyleSheet, ABORT_STYLESHEET))
```
Comment (s-aderhold) [inline lcls_tools/common/pydm_tools/displayUtils.py:28]: Is this using the abort stylesheet on purpose? Or is it a copy/paste error?

File: lcls_tools/superconducting/scLinac.py [hunk 2]
```diff
@@ -197,11 +223,19 @@ def issueMoveCommand(self, numSteps):
...
+            
             print("Motor moving", datetime.now())
             sleep(1)
+            if (abs(self.cavity.detune_best_PV.value) > 150000
+                    and self.cavity.freq_stop_pv.value != 400000):
+                self.cavity.set_chirp_range(400000)
         
-        if caget(self.motor_done_pv.pvname) != 1:
+        if self.motor_done_pv.value != 1:
             raise utils.StepperError("Motor not in expected state")
```
Comment (s-aderhold) [inline lcls_tools/superconducting/scLinac.py:239]: Would it make sense to include the CM/cavity number in this message?

File: lcls_tools/superconducting/scLinac.py [hunk 3]
```diff
@@ -197,11 +223,19 @@ def issueMoveCommand(self, numSteps):
         print("Waiting 5s for the motor to start moving")
         sleep(5)
         
-        while caget(self.motor_moving_pv.pvname) == 1:
+        while self.motor_moving_pv.value == 1:
+            if self.abort_flag:
+                self.abort_pv.put(1)
+                raise utils.StepperAbortError(
+                        f"Abort Requested for {self.cavity.cryomodule.name} cavity {self.cavity.number} stepper tuner")
```
Comment (s-aderhold) [inline lcls_tools/superconducting/scLinac.py:230]:
```suggestion
                        f"Abort requested for {self.cavity.cryomodule.name} cavity {self.cavity.number} stepper tuner")
```

File: lcls_tools/common/pydm_tools/displayUtils.py (hunk 1 above)
Comment (lisazacarias) [reply lcls_tools/common/pydm_tools/displayUtils.py:28]: Maybe I should renamed it to `ERROR_STYLESHEET`... An abort felt like a type of error?

File: lcls_tools/superconducting/scLinac.py (hunk 2 above)
Comment (lisazacarias) [reply lcls_tools/superconducting/scLinac.py:239]: Probably

**Review by nneveu — APPROVED** [review]

**Pull request merged**